packages/transport/src/transports/abstractUsb.ts

```diff
@@ -85,56 +85,55 @@ export abstract class AbstractUsbTransport extends AbstractTransport {
-            if (!acquireIntentResponse.success) {
-                if (this.acquirePromise) {
-                    this.acquirePromise.resolve(undefined);
+                if (!acquireIntentResponse.success) {
+                    if (this.acquirePromise) {
+                        this.acquirePromise.resolve(undefined);
+                    }
+                    return this.error({ error: acquireIntentResponse.error });
                 }
-                return this.error({ error: acquireIntentResponse.error });
-            }
-            this.acquiredUnconfirmed[path] = acquireIntentResponse.payload.session;
 
-            const reset = !!input.previous;
-            const openDeviceResult = await this.transportInterface.openDevice(path, reset);
-            if (!openDeviceResult.success) {
-                if (this.listenPromise) {
-                    this.listenPromise[path].reject(new Error(openDeviceResult.error));
+                this.acquiredUnconfirmed[path] = acquireIntentResponse.payload.session;
+
+                const reset = !!input.previous;
+                const openDeviceResult = await this.transportInterface.openDevice(path, reset);
+
+                if (!openDeviceResult.success) {
+                    if (this.listenPromise) {
+                        this.listenPromise[path].reject(new Error(openDeviceResult.error));
+                    }
+                    return openDeviceResult;
                 }
-                return openDeviceResult;
-            }
 
-            this.sessionsClient.acquireDone({ path });
+                this.sessionsClient.acquireDone({ path });
 
-            if (this.acquirePromise) {
-                this.acquirePromise.resolve(undefined);
-            }
+                if (this.acquirePromise) {
+                    this.acquirePromise.resolve(undefined);
+                }
 
-            if (!this.listenPromise[path]) {
-                return this.success(acquireIntentResponse.payload.session);
-            }
+                if (!this.listenPromise[path]) {
+                    return this.success(acquireIntentResponse.payload.session);
+                }
 
-            return this.listenPromise[path].promise
-                .then(sessionId => {
-                    delete this.listenPromise[path];
-                    return this.success(sessionId);
-                })
-                .catch(err => {
-                    delete this.listenPromise[path];
-                    return this.unknownError(err, [
-                        ERRORS.DEVICE_DISCONNECTED_DURING_ACTION,
-                        ERRORS.SESSION_WRONG_PREVIOUS,
-                        ERRORS.DEVICE_NOT_FOUND,
-                        ERRORS.INTERFACE_UNABLE_TO_OPEN_DEVICE,
-                    ]);
-                });
-        });
+                return this.listenPromise[path].promise.then(sessionId => this.success(sessionId));
```

Comment:
If I understand it correctly, one promise now have two execution branches: In the first one, error is consumed and cleaned from `finally` (as I commented above), and in the second one, result is returned or the same error is thrown? Wouldn't it be easier (or at least more similar to the previous code) to just add the finally block to the second branch? Or is it not equivalent?